Mathlib/CategoryTheory/Grothendieck.lean

```diff
@@ -149,6 +153,108 @@ def forget : Grothendieck F ⥤ C where
 
 end
 
+section
+
+variable {G : C ⥤ Cat}
+
+/-- The Grothendieck construction is functorial: a natural transformation `α : F ⟶ G` induces
+a functor `Grothendieck.map : Grothendieck F ⥤ Grothendieck G`.
+-/
+@[simps!]
+def map (α : F ⟶ G) : Grothendieck F ⥤ Grothendieck G where
+  obj X := {
+    base := X.base
+    fiber := (α.app X.base).obj X.fiber
+  }
+  map {X Y} f := {
+    base := f.base
+    fiber := (eqToHom (α.naturality f.base).symm).app X.fiber ≫ (α.app Y.base).map f.fiber
+  }
+  map_id := by
+    intro X
+    dsimp
+    simp [eqToHom_map]
+    congr 1
+    · rw [eqToHom_app, eqToHom_trans]
+  map_comp := by
+    intro X Y Z f g
+    dsimp
+    congr 1
+    erw [comp_fiber f g]
+    simp [Functor.comp_map, ← Category.assoc]
+    apply congrFun (congrArg (. ≫ .) ?_) ((α.app Z.base).map g.fiber)
+    slice_rhs 3 3 =>
+      rw [← Functor.comp_map _ _]
+    have H :=  Functor.congr_hom (α.naturality g.base).symm f.fiber
+    erw [H, eqToHom_app, eqToHom_app]
+    simp [Functor.comp_map]
+    erw [eqToHom_app]
+    simp only [Cat.comp_obj, eqToHom_trans, eqToHom_refl, Category.comp_id]
+
+theorem map_obj {α : F ⟶ G} (X : Grothendieck F) :
+    (Grothendieck.map α).obj X = ⟨X.base, (α.app X.base).obj X.fiber⟩ := by
+  rfl
+
+theorem map_map {α : F ⟶ G} {X Y : Grothendieck F} {f : X ⟶ Y} :
+    (Grothendieck.map α).map f =
+    ⟨f.base, (eqToHom (α.naturality f.base).symm).app X.fiber ≫ (α.app Y.base).map f.fiber⟩ := by
+  rfl
+
+/-- The functor `Grothendieck.map α : Grothendieck F ⥤ Grothendieck G` lies over `C`.-/
+theorem functor_comp_forget {α : F ⟶ G} :
+    Grothendieck.map α ⋙ Grothendieck.forget G = Grothendieck.forget F := by
+  rfl
+
+theorem map_id_eq : map (𝟙 F) = 𝟙 (Cat.of <| Grothendieck <| F) := by
+  fapply Functor.ext
+  · intro X
+    rfl
+  · intro X Y f
+    simp [map_map]
+    congr
+    erw [NatTrans.id_app]
```

Comment:
@dagurtomas: Actually, changing `erw` to `simp` breaks the proof.

Reply:
Is there a way to fix it?

Reply:
It seems `rw [NatTrans.id_app]; simp` also works, but `simp [NatTrans.id_app]` does not.

Reply:
Ok, `rw` is better than `erw`, but I don't know why `simp` doesn't just close the goal

Reply:
yeah, I could not make it work with any of the  `simp` or `simp only` or `simp [-x, y, ...]`. I don't understand what is going on.